sycl-fusion/CMakeLists.txt

```diff
@@ -9,6 +9,10 @@ set(SYCL_JIT_BASE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
 # directories, similar to how clang/CMakeLists.txt does it.
 set(LLVM_SPIRV_INCLUDE_DIRS "${LLVM_MAIN_SRC_DIR}/../llvm-spirv/include")
 
-add_subdirectory(jit-compiler)
-add_subdirectory(passes)
-add_subdirectory(test)
+if(WIN32)
+  message(WARNING "Kernel fusion not yet supported on Windows")
+else(WIN32)
+  add_subdirectory(jit-compiler)
+  add_subdirectory(passes)
+  add_subdirectory(test)
```

Comment:
@sommerlukas, unfortunately, compilation of `jit-compiler` with clang (XCode compiler to be precise) fails.

See https://github.com/intel/llvm/actions/runs/3910984924/jobs/6683827601.
```
/Users/runner/work/llvm/llvm/src/llvm/include/llvm/Support/Casting.h:280:45: error: unused parameter 'f' [-Werror,-Wunused-parameter]
  static inline bool isPossible(const From &f) { return true; }
                                            ^
/Users/runner/work/llvm/llvm/src/llvm/include/llvm/Support/Casting.h:604:41: error: unused parameter 't' [-Werror,-Wunused-parameter]
  static inline bool isPresent(const T &t) { return true; }
                                        ^
2 errors generated.
```

It's a bit surprising, but it looks like LLVM is less strict on enforcing "no warnings" policy.
I think we should avoid mis-compilations due to errors in external code like this.
To be honest, I'm sure how did we end-up with enabling `-Werror` for `jit-compiler` project. I know that we enabled it for `sycl` project by setting `SYCL_ENABLE_WERROR` CMake variable, but it should not impact `jit-compiler`. Could you investigate, please?

Reply:
Hi @bader,

thanks for reporting this. 

The file which causes this error is `jit_compiler.cpp`, which is not part of `sycl-fusion` (i.e., the JIT compiler), but `sycl` itself (lives in `sycl/source/detail`) and connects the SYCL runtime to the JIT compiler. As it is compiled as part of the SYCL runtime, I think it is expected that it is compiled with `-Werror` if `SYCL_ENABLE_WERROR` is set. 

The warnings (which then become errors with `-Werror`) is caused by including `LLVMContext.h` in `JITContext.h`, which in turn is included by `jit_compiler.cpp`. I'll look into how we can avoid including `LLVMContext.h` here by using an opaque pointer or similar.

Reply:
@bader: I've opened https://github.com/intel/llvm/pull/8017 to address this issue. 

With the fix from this PR, I'm able to compile with `SYCL_ENABLE_WERROR=ON` in my local build (Linux, Clang 10).

Reply:
> I'm able to compile with `SYCL_ENABLE_WERROR=ON` in my local build (Linux, Clang 10).

Were you able to reproduce the problem before the fix with your local build (i.e. Linux, Clang 10)?

Reply:
Yes, with `SYCL_ENABLE_WERROR=ON`, my local build failed with the same error message before implementing this fix.